### PR TITLE
Am 588 doc-am APIs consolidated functional testing

### PR DESCRIPTION
**Pull request by shrivsksavit**
### JIRA link (if applicable) ###
https://tools.hmcts.net/jira/browse/AM-588
https://tools.hmcts.net/jira/browse/AM-653

### Change description ###
Consolidated AM code along with functional test automation.

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x] No
```

### CODE REVIEW

File: src/aat/java/uk/gov/hmcts/ccd/datastore/befta/DataStoreTestAutomationAdapter.java [hunk 1]
```diff
@@ -2,15 +2,60 @@
...
 public class DataStoreTestAutomationAdapter extends DefaultTestAutomationAdapter {
 
     private TestDataLoaderToDefinitionStore loader = new TestDataLoaderToDefinitionStore(this);
 
     @Override
     public void doLoadTestData() {
-        loader.addCcdRoles();
-        loader.importDefinitions();
+//        loader.addCcdRoles();
+//        loader.importDefinitions();
```
Comment (aashishaggarwal123) [inline src/aat/java/uk/gov/hmcts/ccd/datastore/befta/DataStoreTestAutomationAdapter.java:17]: uncomment these 2 lines.

Comment (shrivsk) [reply src/aat/java/uk/gov/hmcts/ccd/datastore/befta/DataStoreTestAutomationAdapter.java]: done

Comment (shrivsksavit) [reply src/aat/java/uk/gov/hmcts/ccd/datastore/befta/DataStoreTestAutomationAdapter.java]: done

File: src/aat/java/uk/gov/hmcts/ccd/datastore/befta/DataStoreTestAutomationAdapter.java [hunk 2]
```diff
@@ -13,4 +17,45 @@ public void doLoadTestData() {
...
+        String docAmUrl = EnvironmentVariableUtils.getRequiredVariable("CASE_DOCUMENT_AM_URL");
+        if (key.equals("documentIdInTheResponse")) {
+            try {
+                String href = (String) ReflectionUtils
+                    .deepGetFieldInObject(scenarioContext,
+                                          "testData.actualResponse.body._embedded.documents[0]._links.self.href");
+                return href.substring(href.length() - 36);
+            } catch (Exception exception) {
+//                logger.error("Exception while getting the Document ID from the response :{}", exception.getMessage());
+                return "Error extracting the Document Id";
```
Comment (MSancaktutar) [inline src/aat/java/uk/gov/hmcts/ccd/datastore/befta/DataStoreTestAutomationAdapter.java:31]: throw new FunctionalTestException("Error extracting the Document Id", e);

File: src/aat/resources/features/F-1003/F-1003-V2/F-1003-V2-Base-Root.td.json [hunk 3]
```diff
@@ -0,0 +1,71 @@
+{
+  "_guid_": "F-1003-V2-Base-Root",
+  "users": {
```
Comment (MSancaktutar) [inline src/aat/resources/features/F-1003/F-1003-V2/F-1003-V2-Base-Root.td.json:3]: why not extend this from a common one from framework?

File: src/aat/resources/features/F-1003/F-1003-V3/F-1003-V3.feature [hunk 4]
```diff
@@ -0,0 +1,106 @@
...
+  @S-1031
+  Scenario: must successfully create a case with multiple documents uploded
+    Given a user with [an active caseworker profile in CCD with full permissions on a document field],
+    And   a successful call [to upload a document with mandatory metadata] as in [Default_Document_Upload_Data],
+    And   a successful call [to upload a document with mandatory metadata] as in [Default_Document_Upload_Data_01],
+    And   a successful call [to upload a document with mandatory metadata] as in [Default_Document_Upload_Data_02],
+    And   a successful call [to upload a document with mandatory metadata] as in [Default_Document_Upload_Data_03],
+    And   a successful call [to upload a document with mandatory metadata] as in [Default_Document_Upload_Data_04],
+    And   a successful call [to upload a document with mandatory metadata] as in [Default_Document_Upload_Data_05],
+    And   a successful call [to upload a document with mandatory metadata] as in [Default_Document_Upload_Data_06],
```
Comment (MSancaktutar) [inline src/aat/resources/features/F-1003/F-1003-V3/F-1003-V3.feature:29]: bett Default_Document_Upload_Data_01 to Default_Document_Upload_Data_07

File: src/aat/resources/features/F-1003/F-1003-V3/Default_Document_Upload_Data_01.td.json [hunk 5]
```diff
@@ -0,0 +1,29 @@
...
+        {
+          "key": "jurisdictionId",
+          "value": "BEFTA_JURISDICTION_2"
+        },
+        {
+          "key": "files",
+          "filePath": "features/common/files/Sample.doc"
+        }
+      ]
+    }
```
Comment (MSancaktutar) [inline src/aat/resources/features/F-1003/F-1003-V3/Default_Document_Upload_Data_01.td.json:27]: in fact, there could be a nice way of eliminating all these repeated blocks, and have just the below lines:
```
  "_guid_": "Default_Document_Upload_Data_01",
  "_extends_": "Default_Document_Upload_Data_Base",
```

This is possible, and I can show you how, if you like.
Hint: The file names could be dynamically picked from a custom value with respect to the _guid_ value.

File: src/aat/resources/features/F-1003/F-1003-V3/F-1003-V3.feature (hunk 4 above)
Comment (MSancaktutar) [inline src/aat/resources/features/F-1003/F-1003-V3/F-1003-V3.feature:29]: these can also be managed by using only 1 test data file, each time with a different file - I can show if you like.

**Pull request merged**